src/config/mod.rs

```diff
@@ -1065,4 +1074,217 @@ make_backup = false
             ])
         );
     }
+
+    #[cfg(test)]
+    mod required_version {
+        use super::*;
+
+        #[allow(dead_code)] // Only used in tests
+        fn get_current_version() -> semver::Version {
+            semver::Version::parse(env!("CARGO_PKG_VERSION")).unwrap()
+        }
+
+        #[nightly_only_test]
+        #[test]
+        fn test_required_version_default() {
+            let config = Config::default();
+            assert!(config.version_meets_requirement());
+        }
+
+        #[nightly_only_test]
+        #[test]
+        fn test_current_required_version() {
+            let toml = format!("required_version=\"{}\"", env!("CARGO_PKG_VERSION"));
+            let config = Config::from_toml(&toml, Path::new("")).unwrap();
+
+            assert!(config.version_meets_requirement());
+        }
```

Comment:
To make the new functionality easier to test I propose extracting the semver checking into a standalone function that takes the the `version` and the `required_version` and performs the check. Then you can implement `version_meets_requirement` in terms of this new function. Here's one idea:
```rust
fn semver_version_check(version: &str, required_version: &str) -> bool {
    // parse and check versions here
}
```

Might still be useful to have a few of these tests with `env!("CARGO_PKG_VERSION")` and calling `version_meets_requirement` directly, but I think creating a standalone inner function that does all the heavy lifting will be more flexible in terms of writing tests.